full-service/src/error.rs

```diff
@@ -67,6 +76,18 @@ impl From<TxoServiceError> for WalletServiceError {
     }
 }
 
+impl From<ProofServiceError> for WalletServiceError {
+    fn from(src: ProofServiceError) -> Self {
+        Self::ProofService(src)
+    }
+}
+
+impl From<TransactionLogServiceError> for WalletServiceError {
```

Comment:
Is there a way to reduce all of this duplication of `impl From`? It seems like every error class needs this line. Macros?

Reply:
Great question - I would love to figure out how to reduce the error conversion boilerplate